v2/internal/frontend/desktop/darwin/WailsWebView.m

```diff
@@ -0,0 +1,120 @@
+#import "WailsWebView.h"
+#import "message.h"
+
+
+@implementation WailsWebView
+@synthesize disableWebViewDragAndDrop;
+@synthesize enableDragAndDrop;
+
+- (BOOL)prepareForDragOperation:(id<NSDraggingInfo>)sender
+{
+  if ( !enableDragAndDrop ) {
+    return [super prepareForDragOperation: sender];
+  }
+
+  if ( disableWebViewDragAndDrop ) {
+    return YES;
+  }
+
+  return [super prepareForDragOperation: sender];
+}
+
+- (BOOL)performDragOperation:(id <NSDraggingInfo>)sender
+{
+  if ( !enableDragAndDrop ) {
+    return [super performDragOperation: sender];
+  }
+
+  NSPasteboard *pboard = [sender draggingPasteboard];
+
+  // if no types, then we'll just let the WKWebView handle the drag-n-drop as normal
+  NSArray<NSPasteboardType> * types = [pboard types];
+  if( !types )
+    return [super performDragOperation: sender];
+
+  // getting all NSURL types
+  NSArray<Class> *url_class = @[[NSURL class]];
+  NSDictionary *options = @{};
+  NSArray<NSURL*> *files = [pboard readObjectsForClasses:url_class options:options];
+
+  // collecting all file paths
+  NSString *files_str = @"";
+  NSMutableArray *files_strs = [[NSMutableArray alloc] init];
```

Comment:
Does this need to be released?

Reply:
@APshenkin

Reply:
@leaanthony @jakubpeleska 

Here we iterate over NSURL array to convert them to just strings, then join then all in one string. 

So `files_strs` is needed. Only if someone have better know how to do this mapping, as I'm not Objective-C expert 😅. 

However `files_str` is redundant, as it's not used. You can remove it.

Reply:
I was thinking the `[alloc]` would require a `[release]`

Reply:
Oh, yes you are right, sorry misunderstood you 😅